plugins/calibrecompanion.koplugin/main.lua

```diff
@@ -11,6 +11,44 @@ local T = require("ffi/util").template
 
 require("ffi/zeromq_h")
 
+-- supported formats
+local extensions = require("extensions")
+local function getExtensionPathLengths()
+    local t = {}
+    for _, v in pairs(extensions) do
+        t[v] = 42
+    end
+    return t
+end
+
+-- info from calibre metadata to keep track of books on device.
+local calibreBookId = {
+    "uuid",
+    "lpath",
+    "authors",
+    "title",
+    "last_modified",
+}
+
+-- dump table to file
+local function dumpDb(t, file)
+    if not t or not file or file == "" then return end
+    local dump = require("dump")
+    local f = io.open(file, "w")
+    if f then
+        f:write("return "..dump(t))
+        f:close()
+    end
+end
+
+-- delete file
+local function delete(file)
+    if not file then return end
+    if lfs.attributes(file, "mode") == "file" then
+        os.remove(file)
```

Comment:
With or without .sdr?

Reply:
I would say without. Mainly because:

1. There's no way to sync our metadata with calibre yet.
2. User can restore the book from its calibre library. Book will be named the same, so the same metadata will work again.

Reply:
Alright. :+1: